fix(formatter): handle poor layout for grouped call arguments

Dunqing opened the pull request:
This PR is a follow-up on #16036 

Move `PoorLayout` handling from `Function` and `ArrowFunctionExpression` to call arguments

leaysgur reviewed (approved):
@Dunqing Would you please resolve these conflicts? 🙇🏻  (and also merge it after that)

Dunqing committed:
fix(formatter): handle poor layout for grouped call arguments (#16093)

This PR is a follow-up on #16036

Move `PoorLayout` handling from `Function` and `ArrowFunctionExpression` to call arguments
